service/api/service_impl.go

```diff
@@ -149,6 +151,111 @@ func (s serviceImpl) PublishToLocalQueue(
 	return nil
 }
 
+func (s serviceImpl) Rpc(
+	ctx context.Context, request xdbapi.ProcessExecutionRpcRequest,
+) (response *xdbapi.ProcessExecutionRpcResponse, retErr *ErrorWithStatus) {
+	latestPrcExe, err := s.store.GetLatestProcessExecution(ctx, data_models.GetLatestProcessExecutionRequest{
```

Comment:
I don't think it's safe outside a transaction. The process execution could get terminated or even deleted right after this query.

Reply:
And you cannot put it into transaction because later there will be a API call to the worker. This is similar to how the waitUntil/Execute API works.

Reply:
Then we need to re-design ALL the transactions that query the table `xdb_sys_latest_process_executions` to lock the row in `xdb_sys_latest_process_executions` as the first step in each transation.

Reply:
There will be two places to GetLatestProcessExecution. One is at the beginning of the RPC method, by session, to get the process info to call worker API. Another is after getting the worker response, use tx to get the latest process execution to lock.

Reply:
The GetLatestProcessExecution before worke RPC call will still be in the session. Once getting the worker RPC response, we will directly use the `tx.SelectProcessExecutionForUpdate` to lock.

Reply:
And two other places in doStopProcessTx and doPublishToLocalQueueTx can use sessions to SelectLatestProcessExecution, I don't think there will be race condition. Can you just provide a detailed example to prove the opposite side?